bangc-ops/mlu_op.h

```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
+ * of argmax for each Roi (Regions of Interest) to perform ::mluOpRoiPoolingForward operation.
+ *
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and queues in
+ * ::mluOpRoiPoolingForward operation. For detailed information, see ::mluOpHandle_t.
+ * @param[in] pooling_mode
+ * The pooling mode of ROI Pooling Forward defined in ::mluOpPoolingMode_t.
+ * @param[in] input_desc
+ * The descriptor of the input tensor in the roipoolingforward process. For detailed
+ * information, see ::mluOpTensorDescriptor_t.
+ * @param[in] input
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] rois_desc
+ * The descriptor of the ROIs tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] rois
+ * Pointer to the MLU memory that stores the rois tensor.
+ * @param[in] spatial_scale
+ * The spatial scale of each ROI in the input feature map.
+ * @param[in] output_desc
+ * The descriptor of the output tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[out] output
+ * Pointer to the MLU memory that stores the output tensor.
+ * @param[out] argmax
+ * Pointer to the MLU memory that stores the argmax tensor. This pointer may be NULL. The tensor
+ * \b argmax means input feature index of maximum for each ROI.
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM.
+ *
+ * @par Data Type
+ * - This function supports the following data types for input tensor \b input, rois tensor \b rois,
+ *   output tensor \b output, and argmax tensor \b argmax. The data type of \b input, \b rois, and \b output
+ *   must be the same.
+ *   - input tensor: half, float.
+ *   - rois tensor: half, float.
+ *   - output tensor: half, float.
+ *   - argmax tensor: int32.
+ *
+ * @par Data Layout
+ * - The supported data layout of \b input, \b rois, \b output, \b argmax are as follows:
+ *   - input tensor: \b MLUOP_LAYOUT_NHWC.
+ *   - ROIs tensor: \b MLUOP_LAYOUT_ARRAY, only supports 2D tensor.
+ *   - output tensor: \b MLUOP_LAYOUT_NHWC.
+ *   - argmax tensor: \b MLUOP_LAYOUT_NHWC.
+ *
+ * @par Scale Limitation
+ * - The value of \b pooling_mode only supports \p MLUOP_POOLING_MAX.
+ * - The input tensor, output tensor and argmax tensor must have four dimensions.
+ * - The size of the lowest dimension of input tensors, output tensor and argmax tensor must be the same.
+ * - The total number of dimensions of ROIs tensor must be 2.
+ * - The size of the highest dimension of output tensor and rois tensor must be the same.
+ * - The shape of \b rois must be [rois_num, 5]. \b rois_num means the total number of \b rois.
+ * - \b Rois[i] consists of [batch_id, x1, y1, x2, y2]. \b batch_id should be in the range of
+ * [0, batch_num - 1]. \b x1 and \b y1 should be greater than or equal to 0. \b x2 should be
+ * greater than \b x1. \b y2 should be greater than \b y1. \b batch_id means id of batch of \b rois. \b x1, \b y1,
```

Comment:
建议把 should 批量替换成must，means 换成represents，
```suggestion
 * greater than \b x1. \b y2 must be greater than \b y1. \b batch_id represents ID of the batch of \b rois. \b x1, \b y1,
```